examples/scripts/ppo-sentiment-t5-small.py

```diff
@@ -45,22 +43,34 @@
 config = PPOConfig(
     model_name="lvwerra/t5-imdb",
     learning_rate=5e-5,
-    batch_size=512,
-    forward_batch_size=16
+    batch_size=256,
+    forward_batch_size=16,
 )
-
 # We then define the arguments to pass to the sentiment analysis pipeline.
 # We set `return_all_scores` to True to get the sentiment score for each token.
 sent_kwargs = {
-    "top_k": None,
+    "return_all_scores": True,
     "function_to_apply": "none",
     "batch_size": config.forward_batch_size
 }
-
 # Below is an example function to build the dataset. In our case, we use the IMDB dataset
 # from the `datasets` library. One should customize this function to train the model on
 # its own dataset.
-def build_dataset(tokenizer, dataset_name="imdb", input_min_text_length=2, input_max_text_length=8):
+def build_dataset(config, dataset_name="imdb", input_min_text_length=2, input_max_text_length=8):
+    """
+    Build dataset for training. This builds the dataset from `load_dataset`, one should 
+    customize this function to train the model on its own dataset.
+    
+    Args:
+        dataset_name (`str`): 
+            The name of the dataset to be loaded.
+    
+    Returns:
+        dataloader (`torch.utils.data.DataLoader`):
+            The dataloader for the dataset.
+    """
+    tokenizer = AutoTokenizer.from_pretrained(config.model_name)
+    #tokenizer.pad_token = tokenizer.eos_token
```

Comment:
This change was also on purpose: we don't use config inside the function and we can pass the tokenizer from outside so we don't initialize it twice.

Reply:
Thanks! Makes sense, reverted the change in [d0875b5](https://github.com/lvwerra/trl/pull/93/commits/d0875b5eab233993493d6390be2d2bfe5d2e707a)